### PR TITLE
DOC: Add plot options API examples for Imviz

### LINKED ISSUE #1477: Document how to auto-scale in Imviz for multiple images at once
**Reporter: pllim**

There is a need to add user documentation on how to use https://github.com/spacetelescope/jdaviz/pull/1343 to accomplish #750 .

Desired workflow:

1. Load two images in Imviz.
2. Interactively adjust scale (stretch? percentile?) in an image.
3. Have the settings automatically applied to the other image.

[🐱](https://jira.stsci.edu/browse/JDAT-2670)

 --- 

 DISCLAIMER: This issue was autocreated by the Jdaviz Issue Creation Bot on behalf of the reporter. If any information is incorrect, please contact [Duy Nguyen](https://github.com/duytnguyendtn)

### PR DESCRIPTION
**Pull request by pllim**
### Description

This pull request is to add user facing examples for plot options API for Imviz so users can operate on multiple images at once.

Fixes #1477 

### Change log entry

- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [x] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [x] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Did the CI pass? If not, are the failures related?
- [x] Is a milestone set?
- [x] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

File: docs/imviz/displayimages.rst [hunk 1]
```diff
@@ -280,6 +286,22 @@ Change the equation that is used to convert data values between
 :guilabel:`min` and :guilabel:`max` to the 0 to 1 scale of pixel saturation on the displayed
 image.
 
+From the API
+^^^^^^^^^^^^
+
+To set the stretch function for just the image being displayed.
+The acceptable values are as defined by glue backend::
```
Comment (pllim) [inline docs/imviz/displayimages.rst:293]: @rosteen , glue vs Glue is back.

**Comment by pllim** [comment]
Not sure why change log check is cancelled but this PR does not need one. 🤷

File: docs/imviz/displayimages.rst [hunk 2]
```diff
@@ -280,6 +286,22 @@ Change the equation that is used to convert data values between
...
+
+    viewer = imviz.default_viewer
+    viewer.stretch_options
+    viewer.stretch = 'sqrt'
+
+To set the stretch function for all the images at once.
+The acceptable values are the same as the GUI menu options::
```
Comment (kecnry) [inline docs/imviz/displayimages.rst:300]:
```suggestion
The acceptable values are the same as the GUI menu options
can be accessed with ``plot_options.stretch_function.choices``::
```

File: docs/imviz/displayimages.rst [hunk 3]
```diff
@@ -289,13 +311,34 @@ Min
...
+
+To set the percentile for all the images at once::
+
+    plot_options.select_all()
+
+    # Preset
+    plot_options.stretch_preset = '95%'
+
+    # Custom
+    plot_options.stretch_preset = 'Custom'
```
Comment (kecnry) [inline docs/imviz/displayimages.rst:339]: technically setting the preset to 'Custom' isn't necessary, but I agree its nice to be verbose here to show the user what is happening.

**Review by kecnry — COMMENTED** [review]
If the goal here is to provide examples for both the astrowidgets and plugin APIs, I think the user might need a little more context (without getting into development details) for the distinction between the two so they don't get confused.... but that is a fine line to try to walk.

How would you feel about moving the plot options examples into a "From the API" section in the plugin docs and link to it from here?  This page could then show the astrowidgets API only and include a link that says something like "alternatively, these options can be set from the plot options plugin or its API, which provides the ability to set options across multiple selected viewers/layers simultaneously"?

**Comment by pllim** [comment]
> How would you feel about moving the plot options examples into a "From the API" section

I think this is a better question for @camipacifici or @hcferguson . They would be reading the doc, not me.

**Review by rosteen — APPROVED** [review]
Looks good to me, thanks.

File: docs/imviz/displayimages.rst (hunk 3 above)
Comment (duytnguyendtn) [reply docs/imviz/displayimages.rst:340]: I agree; I usually like examples where they show "a little extra"

File: docs/imviz/displayimages.rst [hunk 4]
```diff
@@ -305,13 +349,17 @@ The spectrum of colors used to visualize data can be changed using this drop dow
...
+To set the colormap for just the image being displayed::
 
     viewer = imviz.default_viewer
-    viewer.cuts = '95%'
     viewer.colormap_options
     viewer.set_colormap('Viridis')
 
+To set the colormap for all the images at once::
+
+    plot_options.select_all()
```
Comment (duytnguyendtn) [inline docs/imviz/displayimages.rst:360]: Same as abovee

```suggestion
    plot_options = imviz.plugins['Plot Options']
    plot_options.select_all()
```

File: docs/imviz/displayimages.rst [hunk 5]
```diff
@@ -280,6 +286,23 @@ Change the equation that is used to convert data values between
...
+
+    viewer = imviz.default_viewer
+    viewer.stretch_options
+    viewer.stretch = 'sqrt'
+
+To set the stretch function for all the images at once.
+The acceptable values are the same as the GUI menu options
+can be accessed with ``plot_options.stretch_function.choices``::
+
+    plot_options.select_all()
```
Comment (duytnguyendtn) [inline docs/imviz/displayimages.rst:303]: Same as other comments 

```suggestion
    plot_options = imviz.plugins['Plot Options']
    plot_options.select_all()
```

**Review by duytnguyendtn — APPROVED** [review]
A small few nitpicks if you wanted to entertain the suggestion. Otherwise, I think it reads fine! Thanks :)

**Comment by kecnry** [comment]
(I still think it would be worthwhile to separate the different APIs and make the distinction more clear, but looks like there are two approvals without that, so we could always reconsider if it seems to cause any user-confusion)

**Comment by pllim** [comment]
@kecnry , I opened a new issue at https://github.com/spacetelescope/jdaviz/issues/1732 as a follow-up.

Thanks, all!

**Pull request merged**